coralproject/talk
# [CORL-2787] quick site search

## What does this PR do?
This PR makes it possible for moderators to search by site name while filtering comments by site in the moderation view.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## Does this PR introduce any new environment variables or feature flags?
No.

## If any indexes were added, were they added to `INDEXES.md`?
No indexes were added or modified, but if you have not added a text index on sites as indicated in INDEXES.md, you will need to do that

(from INDEXES.md)
## 2021-12-17

```
db.sites.createIndex({ tenantID: 1, "$**": "text", createdAt: -1 });
```
  - This index creates a text search on sites so that when you use a text-filter base on site name, Mongo knows how to resolve your search query.
---

## How do I test this PR?
1. (Not required but helpful) create lots of sites
2. As a user with moderation privileges navigate to the moderation queue
3. Click on the Site Selector component in the upper left hand corner
4. Observe that the dropdown opens as usual, and you can scroll through sites and it paginates as needed
5. Observe that the button has turned into a text input
6. Type the first few letters of a site name, wait one second
7. Observe that the dropdown updates and is populated only by sites who's names begin with that prefix
8. Select a site from the dropdown and observe that the menu closes, that site's name is displayed on the button, and comments are filtered to just comments from that site

## Where any tests migrated to React Testing Library?
No.

## How do we deploy this PR?
No special considerations needed.
